### PR TITLE
Refactor audio.c and audio.h

### PR DESCRIPTION
**Pull request by BodrickLight**
No functionality changes, just renaming methods and variables, deindenting, and some slight refactoring.

### CODE REVIEW

File: src/audio/audio.c [hunk 1]
```diff
@@ -94,69 +99,86 @@ void audio_get_devices()
...
+	int volume = 0;
+	if (ebx == 0x8001) {
+		int volumeDown = 0;
+		rct_map_element *element = map_get_surface_element_at(x / 32, y / 32);
+		if (element && (element->base_height * 8) - 5 > z)
+			volumeDown = 10;
+
+		sint16 rx;
+		sint16 ry;
+		switch (get_current_rotation()) {
```
Comment (duncanspumpkin) [inline src/audio/audio.c]: This is a `coordinate_3d_to_2d` re implementation.

File: src/audio/audio.c [hunk 2]
```diff
@@ -164,49 +186,53 @@ int sound_play_panned(int sound_id, int ebx, sint16 x, sint16 y, sint16 z)
 *
 *  rct2: 0x006BD0F8
 */
-void start_title_music()
+void audio_start_title_music()
 {
-	int musicPathId;
-	switch (gConfigSound.title_music) {
-	default:
+	if (gGameSoundsOff || !RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) & SCREEN_FLAGS_TITLE_DEMO) {
```
Comment (duncanspumpkin) [inline src/audio/audio.c]: You are missing the brackets on the second statement so I think the '!' will be on the screen flags instead of the the result.

**Comment by duncanspumpkin** [comment]
Great work.

File: src/audio/audio.c [hunk 3]
```diff
@@ -94,69 +99,86 @@ void audio_get_devices()
...
-					if (vy < 0 || vy >= viewport->view_height || vx < 0 || vx >= viewport->view_width || volume < -10000) {
-						return sound_id;
-					}
-				}
-			}
+	if (gGameSoundsOff)
+		return 0;
+
+	int volume = 0;
+	if (ebx == 0x8001) {
```
Comment (IntelOrca) [inline src/audio/audio.c]: ebx is the pan with a range from 0x8000 (-32768) to 7FFF (32767).
It has two special values, 0x8000 and 0x8001 which should be defined as named constants.

Its not particularly nice, but we have to keep it like this until we know we have implemented all play sound callers.

**Comment by BodrickLight** [comment]
I should have corrected those issues now. I refactored audio_sound_play_panned a bit, but it should have exactly the same behaviour as it did before.

File: src/peep/peep.c [hunk 4]
```diff
@@ -7851,7 +7851,7 @@ static void peep_on_exit_ride(rct_peep *peep, int rideIndex)
 
 		int laugh = scenario_rand() & 7;
 		if (laugh < 3) {
-			sound_play_panned(SOUND_LAUGH_1 + laugh, 0x8001, peep->x, peep->y, peep->z);
+			audio_play_sound_panned(SOUND_LAUGH_1 + laugh, 0x8001, peep->x, peep->y, peep->z);
```
Comment (zsilencer) [inline src/peep/peep.c]: looks like you may have missed this one

**Pull request merged**